### PR TITLE
Create ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md

### PR DESCRIPTION
**Pull request by gen-App-Enablement**
Release note created by bode

Ticket: [DM-3308](https://cumulocity.atlassian.net/browse/DM-3308)
Original PR: [6327](https://github.softwareag.com/IOTA/cumulocity-ui/pull/6327)

### CODE REVIEW

File: content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+date: ""
+title: Fix type error when using `hookDeviceListColumns` with factory
```
Comment (gen-App-Enablement) [inline content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md:3]: The Claude AI suggested to use the following title:
```suggestion
title: Fixed type error when using hookDeviceListColumns with factory
```

File: content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+component:
+  - value: component--KIsStyzM
+    label: Device Management app
+build_artifact:
+  - value: tc-pjJiURv9Y
+    label: ui-c8y
+ticket: DM-3308
+version: 1019.24.6
+---
+Fix type error when using `hookDeviceListColumns` with factory
```
Comment (gen-App-Enablement) [inline content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md:17]: The Claude AI suggested to use the following body:
```suggestion
In the Cumulocity IoT platform, the hookDeviceListColumns API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, preventing the customization from working as expected. This issue has now been resolved. With this fix, users can successfully utilize the hookDeviceListColumns API in combination with a factory to customize the device list columns according to their requirements without encountering any type errors.
```

**Review by gen-App-Enablement — COMMENTED** [review]
AI improvements

File: content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md (hunk 2 above)
Comment (bozhidardedov) [inline content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md:17]:
```suggestion
In the Cumulocity IoT platform, the hookDeviceListColumns API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, and explicit type casting was needed. This issue has now been resolved.
```

File: content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+component:
+  - value: component--KIsStyzM
+    label: Device Management app
+build_artifact:
+  - value: tc-pjJiURv9Y
+    label: ui-c8y
+ticket: DM-3308
+version: 1019.24.6
+---
+In Cumulocity IoT, the ‘hookDeviceListColumns‘ API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, and explicit type casting was needed. This issue has now been resolved.
```
Comment (BeateRixen) [inline content/change-logs/device-management/ui-c8y-1019-24-6-fix-type-error-when-using-hook-device-list-columns-with-factory.md:17]:
```suggestion
In Cumulocity IoT, the `hookDeviceListColumns` API allows customizing the columns displayed in the device list. However, when using this API with a factory, a type error occurred, and explicit type casting was needed. This issue has now been resolved.
```

**Review by BeateRixen — APPROVED** [review]

**Comment by BeateRixen** [comment]
@l2p- Please review and approve.

**Review by dafkk — APPROVED** [review]

**Pull request merged**